### PR TITLE
Added a String Permutation program using C  

### PR DESCRIPTION
**Pull request by amit14mitra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: C/other/String_permutation.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Added a permutation of strings program using C
```
Comment (atarax665) [inline C/other/String_permutation.c:1]: ```#``` indicates pre=processor directives in C++, use ```//``` for commenting.

File: C/other/String_permutation.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        i++;
+        j--;
+    }
+    return 1;
+}
+
+int main()
+{
+	char **s;
+	int n;
```
Comment (atarax665) [inline C/other/String_permutation.c:34]: Prompt user to input the length of the string.

File: C/other/String_permutation.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	scanf("%d", &n);
```
Comment (atarax665) [inline C/other/String_permutation.c:35]: Prompt user to input the string.

File: C/other/String_permutation.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	do
+	{
+		for (int i = 0; i < n; i++)
+			printf("%s%c", s[i], i == n - 1 ? '\n' : ' ');
+	} while (next_permutation(n, s));
+	for (int i = 0; i < n; i++)
+		free(s[i]);
+	free(s);
+	return 0;
+}
```
Comment (atarax665) [inline C/other/String_permutation.c:51]: Add sample input/output and complexity of the program in commented form. (Refer other merged PRs for insight)

**Review by atarax665 — CHANGES_REQUESTED** [review]
Use of proper comments enhance the readability of the code, so use comments wherever necessary to explain your approach.

**Comment by amit14mitra** [comment]
Hi,

I have done the changes now. you can check it.

Thanks,
Amit

**Comment by atarax665** [comment]
@amit14mitra Sorry, but I couldn't see any new commit.

**Comment by amit14mitra** [comment]
Hi,

I have already PR a totally new file of the same problem with the revised
version.

Thanks,
Amit

**Pull request closed without merging**